sql/numbertype.go

```diff
@@ -1024,6 +1030,23 @@ func convertToFloat64(t numberTypeImpl, v interface{}) (float64, error) {
 	}
 }
 
+func getFirstLongestNumberValue(v string) string {
```

Comment:
(minor) this implementation looks correct to me, but it might be a little nicer to use more of the regex to help you... currently, this logic is a mix of using a regex and some custom parsing.

If you try using a little more regex, you can probably simplify this code a little bit...

```
numre = regexp.MustCompile("^[ ]*[0-9]*\.?[0-9]+")
```

The regex above anchors to the start of the string, allows for empty spaces before the digit (mysql seems to allow this, too), and also allows for the digits left of the decimal point to be optional (which mysql also supports). https://regex101.com/ was the website I used to quickly test it out, too. 

It would be good to add a couple more tests for those edge cases with number parsing (e.g. "    100.foo" => 100, ".123" => 0.123).

Then, in this method, it might be as simple as: 

```
return numre.FindString(s)
```